src/UbidotsFONA.cpp

```diff
@@ -0,0 +1,521 @@
+#endif
+        _currentValue = 0;
+        return false;
+    }
+    fonaSS.println(F("AT+CIPCLOSE"));
+    if (strstr(readData(4000), "CLOSE OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPCLOSE"));
+#endif
+        _currentValue = 0;
+        return false;
+    }
+    fonaSS.println(F("AT+CIPSHUT"));
+    if (strstr(readData(4000), "SHUT OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSHUT"));
+#endif
+        _currentValue = 0;
+        return false;
+    }
+    _currentValue = 0;
+    free(allData);
+    return true;
+}
+
+/***************************************************************************
+FUNCTIONS TO RETRIEVE DATA
+***************************************************************************/
+
+/**
+ * This function is to get value from the Ubidots API with the device label
+ * and variable label
+ * @arg device_label is the label of the device
+ * @arg variable_label is the label of the variable
+ * @return num the data that you get from the Ubidots API
+ */
+float Ubidots::getValue(char* device_label, char* variable_label) {
+    char* serverResponse;
+    char* response;
+    float num;
+
+    char* allData = (char *) malloc(sizeof(char) * 700);
+    sprintf(allData, "%s/%s|LV|%s|%s:%s|end", USER_AGENT, VERSION, _token, device_label, variable_label);
+
+    fonaSS.println(F("AT+CIPMUX=0"));
+    if (strstr(readData(4000), "OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPMUX"));
+#endif
+        return false;
+    }
+    fonaSS.print(F("AT+CIPSTART=\"TCP\",\""));
+    fonaSS.print(_server);
+    fonaSS.print(F("\",\""));
+    fonaSS.print(PORT);
+    fonaSS.println(F("\""));
+    if (strstr(readData(4000), "CONNECT OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSTART"));
+#endif
+        return false;
+    }
+    fonaSS.print(F("AT+CIPSEND="));
+    fonaSS.println(dataLen(allData));
+    if (strstr(readData(4000), ">") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSEND"));
+#endif
+        return false;
+    }
+    fonaSS.println(allData);
+    char* resp = readData(4000);
+    response = strtok(resp, "|");
+    serverResponse = response;
+    while (response!=NULL) {
+        printf("%s\n", response);
+        response = strtok(NULL, "|");
+        if (response != NULL) {
+            serverResponse = response;
+        }
+    }
+    num = atof(serverResponse);
+
+    fonaSS.println(F("AT+CIPCLOSE"));
+    if (strstr(readData(4000), "CLOSE OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error sending data"));
+#endif
+        return false;
+    }
+    fonaSS.println(F("AT+CIPSHUT"));
+    if (strstr(readData(4000), "SHUT OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSHUT"));
+#endif
+        return false;
+    }
+    free(allData);
+    return num;
+}
+
+int Ubidots::readLine(uint32_t ts_max, bool multiline) {
```

Comment:
Where are you using this method? I see it declared but it is not being used anywhere